### PR TITLE
manage Falcon ASGI

### PR DESCRIPTION
**Pull request by maysse17**

### CODE REVIEW

File: .vscode/settings.json [hunk 1]
```diff
@@ -0,0 +1,3 @@
+{
```
Comment (rdidyk) [inline .vscode/settings.json]: please remove IDE specific files from repo and add `.vscode/` pattern to the `.gitignore` file

Comment (maysse17) [reply .vscode/settings.json]: I removed IDE files and I added .vscode to .gitignore

**Comment by rdidyk** [comment]
Hi @maysse17 
thank you for your contribution to this project, I appreciate it
however, I wouldn't mix sync and async approaches here for this library due to some complications it might introduce
for instance: in this library, we're using `jinja2.FileSystemLoader` as a templates loader/renderer, that library is working with files IO
thus every call to the `.render()` method will block the asyncio event loop (and all other waiting coroutines) which makes the async approach pointless in this particular case
I think that a better option will be creating a separate library to address an asynchronous approach (using the async version for particular libraries, using `loop.run_in_executor` in cases of incompatible libraries, and so on)

**Comment by maysse17** [comment]
yes, I understand your comment and I am agree.
So if you are OK, I can start new library for asynchronous approch based on your code.
tell me If you are OK or not if I can start this work.
Thank you
BR

**Comment by rdidyk** [comment]
no problem mate, you can go with it 👍🏼

**Comment by maysse17** [comment]
thank you

**Comment by rdidyk** [comment]
We decided to implement a separate async. version of library
please follow @maysse17 for any updates and news

**Pull request closed without merging**